mmdet3d/models/roi_heads/mask_heads/primitive_head.py

```diff
@@ -0,0 +1,800 @@
+                    center = points.new_tensor([
+                        torch.mean(coords[selected][:, 0]),
+                        torch.mean(coords[selected][:, 1]), (zmin + zmax) / 2.0
+                    ])
+                    sel_global = indices[selected]
+                    point_mask[sel_global] = 1.0
+                    point_sem[sel_global] = points.new_tensor([
+                        center[0], center[1], center[2], zmax - zmin,
+                        (pts_semantic_mask[indices][0])
+                    ])
+                    point_offset[sel_global] = center - coords[selected]
+
+            # Get the boundary points here
+            point2plane_dist = torch.abs(
+                torch.sum(coords * plane_right[:3], 1) + plane_right[-1])
+            min_dist = point2plane_dist.min()
+            selected = torch.abs(point2plane_dist -
+                                 min_dist) < self.train_cfg['dist_thresh']
+
+            if self.primitive_mode == 'line':
+                _, _, line_sel1, line_sel2 = self.match_point2line(
+                    coords[selected], xmin, xmax, ymin, ymax)
+                for idx, line_select in enumerate([line_sel1, line_sel2]):
+                    if torch.sum(line_select) > \
+                            self.train_cfg['num_point_line']:
+                        point_mask[indices[selected][line_select]] = 1.0
+                        line_center = torch.mean(
+                            coords[selected][line_select], axis=0)
+                        line_center[2] = (zmin + zmax) / 2.0
+                        point_offset[indices[selected][line_select]] = \
+                            line_center - coords[selected][line_select]
+                        point_sem[indices[selected][line_select]] = \
+                            points.new_tensor([line_center[0], line_center[1],
+                                               line_center[2],
+                                               pts_semantic_mask[indices][0]])
+
+            if self.primitive_mode == 'xy':
+                if torch.sum(selected) > self.train_cfg['num_point'] and \
+                        torch.var(point2plane_dist[selected]) < \
+                        self.train_cfg['var_thresh']:
+                    center = points.new_tensor([
+                        torch.mean(coords[selected][:, 0]),
+                        torch.mean(coords[selected][:, 1]), (zmin + zmax) / 2.0
+                    ])
+                    sel_global = indices[selected]
+                    point_mask[sel_global] = 1.0
+                    point_sem[sel_global] = points.new_tensor([
+                        center[0], center[1], center[2], zmax - zmin,
+                        (pts_semantic_mask[indices][0])
+                    ])
+                    point_offset[sel_global] = center - coords[selected]
+
+            # Get the boundary points here
+            vec1 = cur_corners[0] - cur_corners[4]
```

Comment:
Several lines of code are duplicated, can we write smaller functions to reduce repeated code?

For example
```python
            vec1 = cur_corners[0] - cur_corners[4]
            vec2 = cur_corners[4] - cur_corners[5]
            surface_norm = torch.cross(vec1, vec2)
            surface_dis = -torch.dot(surface_norm, cur_corners[5])
```